### PR TITLE
weko#23459: add show and hide role in workflow

### PR DESCRIPTION
**Pull request by weko3-dev22**

### CODE REVIEW

File: modules/weko-workflow/weko_workflow/utils.py [hunk 1]
```diff
@@ -1525,3 +1528,93 @@ def update_indexes_public_state(item_id):
                 index.public_state = True
     if update_db:
         db.session.commit()
+
+
+def get_account_info(user_id):
```
Comment (weko3-dev23) [inline modules/weko-workflow/weko_workflow/utils.py:1533]: This function is duplicate with Line 1428: def get_account_info(user_id).

File: modules/weko-workflow/weko_workflow/utils.py [hunk 2]
```diff
@@ -1525,3 +1528,93 @@ def update_indexes_public_state(item_id):
...
+    """
+    data = get_user_profile_info(user_id)
+    if data:
+        return data.get('subitem_mail_address'), \
+            data.get('subitem_displayname')
+    else:
+        return None, None
+
+
+def get_name_display_hide(list_hide, role):
```
Comment (weko3-dev23) [inline modules/weko-workflow/weko_workflow/utils.py:1548]: This function don't have re-usability, so you can put it a @staticmethod of FlowSettingView class.

Thanks,

File: modules/weko-workflow/weko_workflow/utils.py [hunk 3]
```diff
@@ -1525,3 +1528,93 @@ def update_indexes_public_state(item_id):
...
+                tmp = list_hide.pop(0)
+                wfrole = dict(
+                    workflow_id=wf_id,
+                    role_id=tmp
+                )
+                db.session.execute(WorkflowRole.__table__.insert(), wfrole)
+    db.session.commit()
+
+
+def get_workflows(workflows):
```
Comment (weko3-dev23) [inline modules/weko-workflow/weko_workflow/utils.py:1600]: This function can be a method of class Workflow.

Thanks,

Comment (weko3-dev22) [reply modules/weko-workflow/weko_workflow/utils.py]: Has been fixed.

File: modules/weko-workflow/weko_workflow/utils.py (hunk 1 above)
Comment (weko3-dev22) [reply modules/weko-workflow/weko_workflow/utils.py]: Deleted it.

File: modules/weko-workflow/weko_workflow/admin.py [hunk 4]
```diff
@@ -162,6 +165,20 @@ def index(self):
...
         workflow = WorkFlow()
         workflows = workflow.get_workflow_list()
+        role = Role.query.all()
+        for wf in workflows:
+            hide = Role.query.outerjoin(WorkflowRole) \
+                .filter(WorkflowRole.workflow_id == wf.id) \
+                .filter(WorkflowRole.role_id == Role.id) \
+                .all()
+            if hide:
+                display, hide = get_name_display_hide(hide, role)
```
[the remark this answers is not in the document]
Comment (weko3-dev22) [reply modules/weko-workflow/weko_workflow/admin.py]: Has been fixed.

File: modules/weko-workflow/weko_workflow/utils.py (hunk 2 above)
Comment (weko3-dev22) [reply modules/weko-workflow/weko_workflow/utils.py]: Has been fixed.

**Pull request closed without merging**